### PR TITLE
Fix the ability of the build script to use PR builds in installers

### PR DESCRIPTION
**Pull request by shartte**
Sadly, we currently have no API to access the repositories declared in settings.gradle
When accessing the project repositories, they're just empty, so I opted to use an internal Gradle class here which overall *seems* benign enough. I also linked the corresponding Gradle issue for fixing this missing API at the use location.

Not fixing this complicates the use of PRs in builds since one would have to copy all repository declarations into the neoforge subproject, and then add the PR repository there.

### CODE REVIEW

File: buildSrc/src/main/java/net/neoforged/neodev/installer/LibraryCollector.java [hunk 1]
```diff
@@ -37,11 +40,19 @@ public static List<Library> resolveLibraries(List<URI> repositoryUrls, Collectio
         var result = collector.libraries.stream().map(future -> {
             try {
                 return future.get();
+            } catch (ExecutionException e) {
```
Comment (Technici4n) [inline buildSrc/src/main/java/net/neoforged/neodev/installer/LibraryCollector.java]: I am curious as to why this is better than just throwing a new runtime exception in all cases?

Comment (shartte) [reply buildSrc/src/main/java/net/neoforged/neodev/installer/LibraryCollector.java]: It just makes the log cleaner, honestly. The real "fix" was further down below, fixing the error message reading

"java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: ..."

Comment (shartte) [reply buildSrc/src/main/java/net/neoforged/neodev/installer/LibraryCollector.java]: We can remove this part if you'd find it cleaner

Comment (Technici4n) [reply buildSrc/src/main/java/net/neoforged/neodev/installer/LibraryCollector.java]: Yes, remove it. Either we do everywhere or we do it nowhere.

File: buildSrc/src/main/java/net/neoforged/neodev/NeoDevPlugin.java [hunk 2]
```diff
@@ -467,6 +449,29 @@ public void apply(Project project) {
...
+     * Get the list of Maven repositories that may contain artifacts for the installer.
+     */
+    private static Provider<List<URI>> getInstallerRepositoryUrls(Project project) {
+        return project.provider(() -> {
+            List<URI> repos = new ArrayList<>();
+            var repositories = project.getRepositories();
+            if (repositories.isEmpty()) {
+                // If no project repos are defined, check dependency management on the settings plugin,
+                // which sadly is inaccessible without internals: https://github.com/gradle/gradle/issues/27260
+                repositories = ((GradleInternal) project.getGradle()).getSettings().getDependencyResolutionManagement().getRepositories();
```
Comment (lukebemish) [inline buildSrc/src/main/java/net/neoforged/neodev/NeoDevPlugin.java]: Absolutely no reason to use gradle internal stuff here -- this is just for neodev, you can expose that in the settings plugin to the individual projects with a lifecycle beforeProject listener or something without too much trouble, I'd think.

**Review by lukebemish — CHANGES_REQUESTED** [review]

**Review by Technici4n — APPROVED** [review]

**Pull request merged**